src/components/policy/src/policy/src/policy_manager_impl.cc

```diff
@@ -69,18 +69,44 @@ void PolicyManagerImpl::set_listener(PolicyListener* listener) {
   update_status_manager_.set_listener(listener);
 }
 
+#ifdef USE_HMI_PTU_DECRYPTION
+
 utils::SharedPtr<policy_table::Table> PolicyManagerImpl::Parse(
   const BinaryMessage& pt_content) {
   std::string json(pt_content.begin(), pt_content.end());
   Json::Value value;
   Json::Reader reader;
   if (reader.parse(json.c_str(), value)) {
-    return new policy_table::Table(&value);
+      return new policy_table::Table(&value);
+  } else {
+    return utils::SharedPtr<policy_table::Table>();
+  } 
+}
+
+#else
+
+utils::SharedPtr<policy_table::Table> PolicyManagerImpl::ParseArray(
+  const BinaryMessage& pt_content) {
+  std::string json(pt_content.begin(), pt_content.end());
+  Json::Value value;
+  Json::Reader reader;
+  if (reader.parse(json.c_str(), value)) {
+    //For PT Update received from SDL Server.
+    if(value["data"].size()!=0){
```

Comment:
We're using google coding style. Please refer to it in order to fix indentations